File: local-utils/formatEdgeConfigCiks.js
import fs from 'fs/promises';
import path from 'path';
// get reit file from https://stockmarketmba.com/listofreits.php if it still exists
const REIT_CSV_LOCATION = path.resolve('/Users/liammurphy/Downloads/reits.csv');
const CIK_ENDPOINT = 'https://sec.gov/files/company_tickers.json';
async function formatEdgeConfigCiks() {
	const reitSet = new Set();
	const reitCsv = await fs.readFile(REIT_CSV_LOCATION, 'utf-8');
	for (const line of reitCsv.split('\n').slice(1)) {
		const [ticker] = line.split(',');
		reitSet.add(ticker.substring(1, ticker.length - 1));
	}
	const res = await fetch(CIK_ENDPOINT);
	const data = await res.json();
	console.log('{');
	for (const key of Object.keys(data)) {
		const { cik_str, ticker } = data[key];
		if (!reitSet.has(ticker)) continue;
		console.log(`    "${ticker}": "${cik_str}",`);
	}
	console.log('}');
}
formatEdgeConfigCiks();

File: python_scripts_housefire/src/formatEdgeConfigCiks.py
import requests as r
import pandas as pd
from dotenv import load_dotenv
import os

load_dotenv()

TEMP_DIR_PATH = os.getenv("TEMP_DIR")
if TEMP_DIR_PATH is None:
    raise Exception("TEMP_DIR environment variable not set")
REIT_CSV_PATH = os.path.join(TEMP_DIR_PATH, "reits.csv")

CIK_ENDPOINT = "https://sec.gov/files/company_tickers.json"


def format_edge_config_ciks():
    to_concat_list = ["{"]
    reit_csv = pd.read_csv(REIT_CSV_PATH)
    reit_set = set(reit_csv["Symbol"])
    cik_res = r.get(CIK_ENDPOINT)
    cik_data = cik_res.json()
    for key in cik_data:
        cik_str, ticker = str(cik_data[key]["cik_str"]), cik_data[key]["ticker"]
        if ticker not in reit_set:
            continue
        cik_str = cik_str.zfill(10)
        to_concat_list.append(f'     "{ticker}": "{cik_str}",')
    to_concat_list.append("}")
    return "\n".join(to_concat_list)


if __name__ == "__main__":
    if not os.path.exists(REIT_CSV_PATH):
        raise Exception(f"{REIT_CSV_PATH} does not exist")
    formatted_ciks = format_edge_config_ciks()
    print(formatted_ciks)
